test/tests/functional/pbs_hook_execjob_prologue.py

```diff
@@ -247,12 +250,20 @@ def test_prologue_hook_does_not_execute_twice_on_pbsdsh(self):
         attr = {'event': 'execjob_prologue'}
         self.server.create_import_hook(hook_name, attr, hook_body)
 
-        j = Job(TEST_USER, {'Resource_List.select': '2:ncpus=1',
-                            'Resource_List.place': 'scatter'})
-        j.create_script('pbsdsh -n 1 hostname')
+        j = Job(PBSROOT_USER, {'Resource_List.select': '2:ncpus=1',
+                               'Resource_List.place': 'scatter'})
+        j.create_script('pbsdsh  hostname')
         jid = self.server.submit(j)
+        attribs = self.server.status(JOB, id=jid)
         self.server.expect(JOB, 'queue', op=UNSET, id=jid)
-        match = self.momB.log_match("Job;%s;executed prologue hook" % jid,
-                                    allmatch=True)
-        self.assertTrue(match is not None)
-        self.assertTrue(len(match) == 1)
+        host, opath = attribs[0]['Output_Path'].split(':', 1)
+        ret = self.du.cat(hostname=self.mom.shortname, filename=opath,
+                          sudo=True)
+        mom1 = ret['out'][0].split(".")[0]
```

Comment:
I believe you also need to check if above cat command is successful or output is not empty before reading from the file.

Reply:
I've put a check for cat command as suggested.

Reply:
Thanks Vishrut. what if the hostnames are not in FQDN but short hostname?